### PR TITLE
tweak: expand definition of suggested collision naming convention

### PR DESCRIPTION
**Pull request by devtooligan**
The [current style guide](https://github.com/ethereum/solidity/blob/v0.8.11/docs/style-guide.rst#avoiding-naming-collisions) suggests the use of trailing _ when the desired name of a function or variable collides with a built-in or otherwise reserved word.

However, this convention is widely used to not only apply to reserved words but also to include cases when the name would shadow an existing function or state variable name.

This pr attempts to expand the suggestion to include collisions with existing function or state variable names.

### CODE REVIEW

**Comment by fulldecent** [comment]
This style policy is great.

But please update this PR to be a "DRAFT" on GitHub. I'm hoping we can address new opinions like this AFTER other cleanup work.

**Comment by leonardoalt** [comment]
Since it's a small change, I'd suggest we merge it.

**Comment by Marenz** [comment]
Yeah, I think this is minor enough to just go ahead

**Comment by fulldecent** [comment]
Why save all the easy decisions for the existing team?

By presenting this easy one and some other hards ones together when we are ready for broader input, the process will be more engaging.

We need engagement because our goal here is not just to publish a document but to clean up how people write Solidity.

**Comment by leonardoalt** [comment]
> Why save all the easy decisions for the existing team?

Not sure I get this question.

Larger modifications at once are always harder and take a long time to go through/converge.
Small changes on the other hand can be quickly analyzed and accepted/rejected. What's the problem with accepting this now and if the larger change wants to change it we check it there too?

**Comment by chriseth** [comment]
Please make sure that you keep the lines in the documentation source short. This way diffs are easier to read and conflicts are less likely.

**Comment by fulldecent** [comment]
There's no problem accepting this. It makes sense in this repo.

---

But I'm proposing we can have a few PR's like this ready to go (not one big PR) and wait for merge.

Then we (I) can present the list of them to the community or ask specific people for feedback all at once as a checklist.

The end result is we have a nice way to ask new people for their opinions and it is really easy for them to give their feedback.

I think people might appreciate being asked for that feedback—all at once, with some simple questions like this in there. And if we do it that way we might generate some goodwill and more people might read and use this document.

**Comment by cameel** [comment]
I think I see what you're trying to do. Having some specific points to draw in people sounds like a good idea overall, just not sure if it's really necessary to have these sit in unmerged PRs. Small ones are small enough to either merge them quickly or move them to an issue. And doing big ones before they're accepted might be a wasted effort. I think discussing these in issues first might be better.

But anyway, I think it's important that you have full freedom in how you want to organize the process so my personal vote would be to just let you do it the way you want it and see how it goes :)

BTW, I just created a `style guide` label that can be applied to issues and PRs related to this. Actually, do you have enough permissions in the repo to use labels?

**Comment by devtooligan** [comment]
Hey @cameel @fulldecent @leonardoalt I am looking forward to submitting many prs.  Would it be possible to add me as a `contributor`?  @fulldecent tried to add me but it didn't go through (I think that's why I couldn't tag you as a reviewer in the pr's, @fulldecent!)

**Comment by fulldecent** [comment]
@devtooligan Please let's discuss this on the project board I made at https://github.com/users/fulldecent/projects/2

I hope you and I can discuss some things there before we bring them here to the Solidity repo.

---

I do not have access to use labels here. Thank you for creating the label.

**Comment by cameel** [comment]
I think @chriseth has not added you to CODEOWNERS yet as discussed in https://github.com/ethereum/solidity/issues/12614#issuecomment-1034091671. I'll bring that up tomorrow.

**Review by Marenz — APPROVED** [review]

**Pull request merged**
